src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure.Core/CommandBuilders/ASPNetBenchmarks.CommandBuilder.cs

```diff
@@ -72,14 +72,45 @@ public static (string, string) Build(ASPNetBenchmarksConfiguration configuration
                 commandStringBuilder.Append($" --application.options.traceOutput {Path.Combine(configuration.Output.Path, run.Key, (baseConfiguration.Key + "." + collectType)) + traceFileSuffix}");
             }
 
-            commandStringBuilder.Append($" --application.framework net8.0 ");
+            // Add any additional arguments specified.
+            if (!string.IsNullOrEmpty(configuration.benchmark_settings.additional_arguments))
+            {
+                commandStringBuilder.Append($" {configuration.benchmark_settings.additional_arguments} ");
+            }
+
+            // Add the Framework version.
+            string frameworkVersion = "net8.0";
+            // If the framework version at the top level is explicitly stated, use it.
+            if (!string.IsNullOrEmpty(configuration.Environment.framework_version))
+            {
+                frameworkVersion = configuration.Environment.framework_version;
 
-            string corerunToSend = run.Value.corerun.EndsWith("\\") ? run.Value.corerun.Remove(run.Value.corerun.Length - 1) : run.Value.corerun;
-            commandStringBuilder.Append($" --application.options.outputFiles {Path.Combine(Path.GetDirectoryName(corerunToSend), "*.*" )}");
+                // If the framework version is set at the run level, use it.
+                if (!string.IsNullOrEmpty(run.Value.framework_version))
+                {
+                    frameworkVersion = run.Value.framework_version;
+                }
+            }
+            commandStringBuilder.Append($" --application.framework {frameworkVersion} ");
+
+            string artifactsToUpload = run.Value.corerun!;
+
+            // If the corerun specified is a directory, upload the entire directory.
+            // Else, we upload just the file.
+            if (Directory.Exists(run.Value.corerun!))
+            {
+                artifactsToUpload = Path.Combine(artifactsToUpload, "*.*");
+            }
+            commandStringBuilder.Append($" --application.options.outputFiles {artifactsToUpload} ");
 
             // Get the log.
+            // TODO: Specify the path.
```

Comment:
just checking that this isn't completed by the use of `configuration.Output.Path` below